### PR TITLE
Add tick-time analytics

### PR DESCRIPTION
**Pull request by alexsanford**
Building upon #15792.

This PR logs the duration of an iteration of the event loop to `statsd` at a configurable interval (initially configured at 50 logs per second). This will let us know if there is something causing the event loop to take a long time.

## Testing

- Set `boom_analytics_enabled` to `true` in `config/development.json`

- If you have access to the `statsd` analytics created by requests to `https://pixel.wp.com/boom.gif`, you should be able to load any page in Calypso, and see the analytics data come through.

- If you do not have that access, one way to verify this feature is to change the URL in `client/lib/analytics/statsd.js`. Change `https://pixel.wp.com/boom.gif` to `http://calypso.localhost:3000/boom.gif`. Then, on loading any page in Calypso, you should see the `boom.gif` request in the server logs with the correct attributes.

cc @ehg

### CODE REVIEW

File: server/lib/analytics/tick-timer.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	 * of the following event loop. As per the documentation, a setImmediate
+	 * callback scheduled within another setImmediate callback is guaranteed to
+	 * be run in the following iteration of the event loop.
+	 *
+	 * See:
+	 * https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/
+	 * https://nodejs.org/api/timers.html#timers_setimmediate_callback_args
+	 */
+	setImmediate( () => {
+		const startTime = new Date();
```
Comment (ehg) [inline server/lib/analytics/tick-timer.js]: Should we use `hrtime` here?

Comment (alexsanford) [inline server/lib/analytics/tick-timer.js]: @ehg Yeah, that looks like the right tool for the job 👍 

I'm thinking that the extra resolution isn't necessary though. From my local testing, typical timing values are in the range of 50ms to 300ms, so I think reporting with millisecond precision is plenty. Although I suppose reporting in nanoseconds wouldn't hurt.

Thoughts?

**Comment by apeatling** [comment]
@alexsanford What's the status of this one? Still a chance this will be merged?

**Comment by apeatling** [comment]
Closing, please reopen if you decide to merge/need more reviews.

**Comment by ehg** [comment]
Re-opening this, since I think it'll be valuable to debug some `node` performance problems. /cc @seear

File: server/lib/analytics/tick-timer.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+		setImmediate( () => {
+			analytics.statsd.recordTiming(
```
Comment (ehg) [inline server/lib/analytics/tick-timer.js]: This doesn't seem like it should work, given we're [stubbing out `analytics`.](https://github.com/Automattic/wp-calypso/blob/master/webpack.config.node.js#L168)

**Comment by designsimply** [comment]
@seear, I noticed this PR is old and that you were mentioned as someone to check in with about it. Do you think it should remain open because it would be valuable for debugging `node` performance problems or do you think it would be okay to close this issue with an explanation or intent to re-open it at a later date?

**Comment by seear** [comment]
> Do you think it should remain open because it would be valuable for debugging node performance problems or do you think it would be okay to close this issue with an explanation or intent to re-open it at a later date?

Thanks @designsimply. This PR escaped my notice the first time around. It looks like it may be valuable. I've assigned it to myself, and I'll take a closer look over the next few days and either merge something or close it as appropriate.

**Comment by alisterscott** [comment]
This PR needs to be rebased or closed

**Comment by seear** [comment]
Closing due to this not being a current priority.

**Pull request closed without merging**